File: modules/expo-stockfish/ios/ExpoStockfishModule.swift
import ExpoModulesCore

public class ExpoStockfishModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoStockfish")

    Events("stockfish.line")

    AsyncFunction("start") { [weak self] in
      StockfishBridge.shared().start { [weak self] line in
        self?.sendEvent("stockfish.line", ["line": line])
      }
    }

    Function("send") { (command: String) in
      StockfishBridge.shared().send(command)
    }

    AsyncFunction("stop") {
      StockfishBridge.shared().stop()
    }
  }
}

File: modules/expo-stockfish/android/src/main/java/expo/modules/stockfish/ExpoStockfishModule.kt
package expo.modules.stockfish

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoStockfishModule : Module() {
  companion object {
    init { System.loadLibrary("expo_stockfish") }
  }

  private external fun nativeStart()
  private external fun nativeSend(command: String)
  private external fun nativeStop()

  // Called from JNI on the reader thread.
  @Suppress("unused")
  fun emitLine(line: String) {
    sendEvent("stockfish.line", mapOf("line" to line))
  }

  override fun definition() = ModuleDefinition {
    Name("ExpoStockfish")

    Events("stockfish.line")

    AsyncFunction("start") { nativeStart() }
    Function("send") { command: String -> nativeSend(command) }
    AsyncFunction("stop") { nativeStop() }
  }
}
